File: Java/PrimeNumberGeneratorExercise/test/com/png/util/tests/inputValidationTests.java
package com.png.util.tests;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

import com.png.util.*;

class inputValidationTests {
	InputValidation validation;
	
	@BeforeEach
	void init() {
		validation = new InputValidation();
	}
	
	@Test
	void happyPath_InputsSuccessfullyValidated() {
		boolean isValid = validation.validateInputParameters(new String[] { "1", "10" });
		
		assertTrue(isValid);
		
		assertEquals(1, validation.getStartValue());
		assertEquals(10, validation.getEndValue());
	}

	@Test
	void happyPath_InputsDescendingOrder_Valid() {
		boolean isValid = validation.validateInputParameters(new String[] { "10", "1" });
		
		assertTrue(isValid);
		
		assertEquals(10, validation.getStartValue());
		assertEquals(1, validation.getEndValue());
	}
	
	@Test
	void happyPath_NegativeIntegerInputs_Valid() {
		boolean isValid = validation.validateInputParameters(new String[] { "-2", "-3" });
		
		assertTrue(isValid);
		
		assertEquals(-2, validation.getStartValue());
		assertEquals(-3, validation.getEndValue());
	}
	
	@Test
	void negative_NotEnoughArguments() {
		boolean isValid = validation.validateInputParameters(new String[] { "1" });
		
		assertFalse(isValid);
	}
	
	@Test
	void negative_TooManyArguments() {
		boolean isValid = validation.validateInputParameters(new String[] { "1", "2", "3" });
		
		assertFalse(isValid);
	}
	
	@Test
	void negative_StartValueNotInteger() {
		boolean isValid = validation.validateInputParameters(new String[] { "one", "5" });
		
		assertFalse(isValid);
	}
	
	@Test
	void negative_EndValueNotInteger() {
		boolean isValid = validation.validateInputParameters(new String[] { "1", "five" });
		
		assertFalse(isValid);
	}
	
	@Test
	void negative_BothValuesNotIntegers() {
		boolean isValid = validation.validateInputParameters(new String[] { "one", "five" });
		
		assertFalse(isValid);
	}
}

File: C#/PrimeNumberGeneratorTests/InputValidationTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using PrimeNumberGeneratorExercise.Helpers;

namespace PrimeNumberGeneratorTests
{
    [TestClass]
    public class InputValidationTests
    {
        InputValidation validation;

        [TestInitialize]
        public void Initialize()
        {
            validation = new InputValidation();
        }

        [TestMethod]
        public void HappyPath_InputsSuccessfullyValidated()
        {
            var isValid = validation.ValidateInputParameters(new[] { "1", "10" });

            Assert.IsTrue(isValid);

            Assert.AreEqual(1, validation.StartValue);
            Assert.AreEqual(10, validation.EndValue);
        }

        [TestMethod]
        public void HappyPath_InputsDescendingOrder_Valid()
        {
            var isValid = validation.ValidateInputParameters(new[] { "10", "1" });

            Assert.IsTrue(isValid);

            Assert.AreEqual(10, validation.StartValue);
            Assert.AreEqual(1, validation.EndValue);
        }

        [TestMethod]
        public void HappyPath_NegativeIntegerInputs_Valid()
        {
            var isValid = validation.ValidateInputParameters(new[] { "-2", "-3" });

            Assert.IsTrue(isValid);

            Assert.AreEqual(-2, validation.StartValue);
            Assert.AreEqual(-3, validation.EndValue);
        }

        [TestMethod]
        public void Negative_NotEnoughArguments()
        {
            var isValid = validation.ValidateInputParameters(new[] { "1" });

            Assert.IsFalse(isValid);
        }

        [TestMethod]
        public void Negative_TooManyArguments()
        {
            var isValid = validation.ValidateInputParameters(new[] { "1", "2", "3" });

            Assert.IsFalse(isValid);
        }

        [TestMethod]
        public void Negative_StartValueNotInteger()
        {
            var isValid = validation.ValidateInputParameters(new[] { "one", "5" });

            Assert.IsFalse(isValid);
        }

        [TestMethod]
        public void Negative_EndValueNotInteger()
        {
            var isValid = validation.ValidateInputParameters(new[] { "1", "five" });

            Assert.IsFalse(isValid);
        }

        [TestMethod]
        public void Negative_BothValuesNotIntegers()
        {
            var isValid = validation.ValidateInputParameters(new[] { "one", "five" });

            Assert.IsFalse(isValid);
        }
    }
}
